pkg/app/middleware.go

```diff
@@ -21,10 +23,19 @@ func WithMetricsResponderMiddleware(next http.Handler) http.Handler {
 }
 
 // WithHealthMiddleware returns middleware which responds to the /health endpoint
-func WithHealthMiddleware(next http.Handler) http.Handler {
+func WithHealthMiddleware(next http.Handler, threads []*thread.Thread, logger logrus.FieldLogger) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		if r.Method == http.MethodGet && r.URL.Path == "/health" {
-			w.WriteHeader(http.StatusOK)
+			status := http.StatusOK
+			for _, th := range threads {
+				if th.LastRunTimestamp().IsZero() && time.Since(th.LastRunTimestamp()).Minutes() > 5 {
```

Comment:
lets have the timeout configurable to avoid pushing another hotfix in case we will need to adjust it.

Reply:
Please don't forget to add the env var into https://github.com/openshift/assisted-service/blob/master/openshift/template.yaml